script/tool/lib/src/fetch_deps_command.dart

```diff
@@ -79,22 +80,22 @@ class FetchDepsCommand extends PackageLoopingCommand {
     // `pod install` requires having the platform artifacts precached. See
     // https://github.com/flutter/flutter/blob/fb7a763c640d247d090cbb373e4b3a0459ac171b/packages/flutter_tools/bin/podhelper.rb#L47
     // https://github.com/flutter/flutter/blob/fb7a763c640d247d090cbb373e4b3a0459ac171b/packages/flutter_tools/bin/podhelper.rb#L130
-    if (getBoolArg(platformIOS)) {
-      final int exitCode = await processRunner.runAndStream(
+    final bool precacheIos = getBoolArg(platformIOS);
+    final bool precacheMacOS = getBoolArg(platformMacOS);
+    if (precacheIos || precacheMacOS) {
+      final int precacheExitCode = await processRunner.runAndStream(
         flutterCommand,
-        <String>['precache', '--ios'],
+        <String>['precache', '--${precacheIos ? 'ios' : 'macos'}'],
```

Comment:
Although it doesn't seem that any uses of "fetch-deps" does this currently, it is possible for both `getBoolArg(platformIOS)` and `getBoolArg(platformMacOS)` to be true. Perhaps would be better to make it a function and pass in the platform for each?

Reply:
IIRC the tool will accept multiple platforms, in which case this can be:
```dart
<String>[
  'precache',
  if (precacheIOS)
    '--ios',
  if (precacheMacOS)
    '--macos',
],
```